test/common/phantom-driver.js

```diff
@@ -46,7 +46,8 @@ var page = require('webpage').create();
 var sys = require('system');
 var fs = require('fs');
 var clearedStorage = false;
-var messages = "";
+var messages = [];
+var firstMessageIndex = 0;
```

Comment:
The "first" is a bit confusing. I'm not entirely sure when `respond()` gets called (it's hopelessly overloaded even in this file alone), but maybe something like `currentCallMessageIndex`?